axum-login/src/auth.rs

```diff
@@ -73,20 +77,21 @@ where
 }
 
 #[derive(Clone)]
-pub struct AuthService<S, Store, User, Role = ()> {
+pub struct AuthService<S, Store, UserId, User, Role = ()> {
     inner: S,
-    state: AuthState<Store, User, Role>,
+    state: AuthState<Store, UserId, User, Role>,
 }
 
-impl<S, ReqBody, ResBody, Store, User, Role> Service<Request<ReqBody>>
-    for AuthService<S, Store, User, Role>
+impl<S, ReqBody, ResBody, Store, UserId, User, Role> Service<Request<ReqBody>>
+    for AuthService<S, Store, UserId, User, Role>
 where
     S: Service<Request<ReqBody>, Response = Response<ResBody>> + Clone + Send + 'static,
     S::Future: Send + 'static,
     ReqBody: Send + 'static,
     ResBody: Default + Send + 'static,
-    Store: UserStore<Role, User = User>,
-    User: AuthUser<Role>,
+    Store: UserStore<UserId, Role, User = User>,
+    UserId: Clone + Send + Sync + Serialize + DeserializeOwned + 'static,
```

Comment:
Curious about `DeserializeOwned`: is this required by sqlx? How does this apply to other stores?

I'm a little worried about adding `serde` to the crate if it isn't behind a feature flag.